### PR TITLE
[ONEM-30910] [OIPF] Add window.KeyEvent interface with VK keys mapping

### PR DESCRIPTION
**Pull request by piotr-marcinkowski-red**
This commit was based on a similar change that was done here https://github.com/LibertyGlobal/WPEWebKit/pull/82

The original modification had to be adjusted to align with the changes done in these commits:

https://github.com/LibertyGlobal/WPEWebKit/commit/bbe874dcbccdcfd307c2f34572a08344d3598c18

https://github.com/LibertyGlobal/WPEWebKit/commit/7a9cfbc7b0e2c4cd05d6f93325f64db2753c14a7

### CODE REVIEW

**Review by jacek-skiba-red — COMMENTED** [review]
Technically we are adding window.KeyEvent object, so I would change the title of this PR:
[ONEM-30910] [OIPF] Add window.KeyEvent interface with VK keys mapping

File: Source/WebCore/dom/KeyboardEvent.idl [hunk 1]
```diff
@@ -54,6 +55,63 @@
     undefined initKeyboardEvent([AtomString] DOMString type, optional boolean canBubble = false, optional boolean cancelable = false,
         optional WindowProxy? view = null, optional [AtomString] DOMString keyIdentifier = "undefined", optional unsigned long location = 0,
         optional boolean ctrlKey = false, optional boolean altKey = false, optional boolean shiftKey = false, optional boolean metaKey = false, optional boolean altGraphKey = false);
+
```
Comment (jacek-skiba-red) [inline Source/WebCore/dom/KeyboardEvent.idl]: In context of upstreaming, why we are adding those keys definitions to unrelated KeyboardEvent API?

File: Source/WebCore/CMakeLists.txt [hunk 2]
```diff
@@ -1249,6 +1249,7 @@ set(WebCore_NON_SVG_IDL_FILES
     page/UndoItem.idl
     page/UndoManager.idl
     page/VisualViewport.idl
+    page/VkConsts.idl
```
Comment (jacek-skiba-red) [inline Source/WebCore/CMakeLists.txt]: In context of upstreaming we should enable this interface based on build switch, eg: ENABLE_OIPF_VK

File: Source/WebCore/dom/KeyboardEvent.idl (hunk 1 above)
Comment (piotr-marcinkowski-red) [reply Source/WebCore/dom/KeyboardEvent.idl]: Key definitions removed from KeyboardEvent API

File: Source/WebCore/CMakeLists.txt (hunk 2 above)
Comment (piotr-marcinkowski-red) [reply Source/WebCore/CMakeLists.txt]: ENABLE_OIPF_VK compilation flag added

**Review by jacek-skiba-red — APPROVED** [review]

**Pull request merged**